gpii/node_modules/flowManager/src/UntrustedSettingsPutHandler.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * GPII Untrusted Settings Put Handler
+ *
+ * Copyright 2017 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+// Update preferences by first ensuring the client that requests the update action does have the privilege
+// to update, by verifying the access token embedded in the request "Authorization" header.
+fluid.defaults("gpii.flowManager.cloudBased.untrustedSettings.put.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.flowManager.cloudBased.untrustedSettings.put.handleRequest",
+            args: [
+                "{flowManager}.preferencesDataSource",
+                "{request}",
+                "{request}.req.params.userToken",
+                "{request}.req.body",
+                "{gpii.flowManager.cloudBased.oauth2}.userService",
+                "{gpii.flowManager.cloudBased.oauth2}.authGrantFinder"
+            ]
+        }
+    }
+});
+
+gpii.flowManager.cloudBased.untrustedSettings.put.messages = {
+    success: "Successfully updated."
+};
+
+gpii.flowManager.cloudBased.untrustedSettings.put.handleRequest = function (preferencesDataSource, request, userToken, preferences, userService, authGrantFinder) {
+    var authorizationPromise = gpii.oauth2.getAuthorization(request.req, authGrantFinder);
+
+    authorizationPromise.then(function (authorization) {
+        if (authorization && authorization.gpiiToken === userToken && authorization.allowUntrustedSettingsPut) {
+            // TODO: Verify the received preferences with metadata to make sure they are the ones that are allowed to be updated.
+            // This is not supported at the first release of PSP
```

Comment:
@cindyli Could you update the JIRA issue with a note describing that the work to limit/verify which preferences are being modified is not being done under GPII-2511?

https://issues.gpii.net/browse/GPII-2511

Do we have a separate issue for this work? I believe that the code as it is now will allow this endpoint to modify all preferences of the user.

Reply:
I've updated [GPII-2511](https://issues.gpii.net/browse/GPII-2511?focusedCommentId=30425&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-30425) to add the notes and created [a separate JIRA](https://issues.gpii.net/browse/GPII-2820) to keep track of this todo.